frontend/src/scenes/pipeline/sources/SelfManaged.tsx

```diff
@@ -0,0 +1,82 @@
+import { LemonButton } from '@posthog/lemon-ui'
+import { BindLogic, useActions, useValues } from 'kea'
+import { router } from 'kea-router'
+import { PageHeader } from 'lib/components/PageHeader'
+import { useEffect } from 'react'
+import { DatawarehouseTableForm } from 'scenes/data-warehouse/new/DataWarehouseTableForm'
+import { dataWarehouseTableLogic } from 'scenes/data-warehouse/new/dataWarehouseTableLogic'
+import { urls } from 'scenes/urls'
+
+import { DataWarehouseTable } from '~/types'
+
+interface SelfManagedProps {
+    id: string
+}
+
+export const SelfManaged = ({ id }: SelfManagedProps): JSX.Element => {
+    const { table } = useValues(dataWarehouseTableLogic({ id }))
+    const { loadTable, updateTable, resetTable } = useActions(dataWarehouseTableLogic({ id }))
+
+    useEffect(() => {
+        loadTable()
+    }, [loadTable])
+
+    useEffect(() => {
+        return () => {
+            resetTable()
+        }
+    }, [resetTable])
```

Comment:
We often try to avoid using `useEffect` in favour of logics - especially when it comes to data access / data loading. I'd suggest changing this to a simple kea logic. It has [mount/unmount](https://keajs.org/docs/core/events) actions to replicate the logic here. [See frontend coding conventions for more info on why we do this](https://posthog.com/handbook/engineering/conventions/frontend-coding)

Reply:
Thanks. This is a lot better than what I was doing.